test/oli_web/live/select_source_live_test.exs

```diff
@@ -0,0 +1,424 @@
+      |> render_click()
+
+      assert has_element?(view, "h5", "#{s1.title}")
+      refute has_element?(view, "h5", "#{s2.title}")
+
+      view
+      |> element("button[phx-click=\"reset_search\"]")
+      |> render_click()
+
+      assert has_element?(view, "h5", "#{s1.title}")
+      assert has_element?(view, "h5", "#{s2.title}")
+    end
+
+    test "applies sorting", %{conn: conn} do
+      %Publication{project: project} = insert(:publication)
+      insert(:section, %{base_project: project, title: "Testing A"})
+      insert(:section, %{base_project: project, title: "Testing B"})
+
+      {:ok, view, _html} = live(conn, @live_view_lms_instructor_route)
+
+      view
+      |> element("form[phx-change=\"sort\"")
+      |> render_change(%{sort_by: "title"})
+
+      assert view
+             |> element(".card-deck:last-child")
+             |> render() =~ "Testing B"
+
+      view
+      |> element("form[phx-change=\"sort\"")
+      |> render_change(%{sort_by: "title"})
+
+      assert view
+             |> element(".card-deck:last-child")
+             |> render() =~ "Testing A"
+    end
+
+    test "applies paging", %{conn: conn} do
+      %Publication{id: publication_id, project: project} = insert(:publication)
+      [_first_s | tail] = insert_list(21, :section, base_project: project) |> Enum.sort_by(& &1.title)
+      last_s = List.last(tail)
+
+      {:ok, view, _html} = live(conn, @live_view_independent_learner_route)
+
+      view
+      |> element("form[phx-change=\"sort\"")
+      |> render_change(%{sort_by: "title"})
+
+      assert has_element?(view, "a[phx-value-id=\"publication:#{publication_id}\"]")
+      refute has_element?(view, "a[phx-value-id=\"product:#{last_s.id}\"]")
+
+      view
+      |> element("a[phx-click=\"page_change\"]", "2")
+      |> render_click()
+
+      refute has_element?(view, "a[phx-value-id=\"publication:#{publication_id}\"]")
+      assert has_element?(view, "a[phx-value-id=\"product:#{last_s.id}\"]")
+    end
+
+    test "selects one section correctly", %{conn: conn} do
+      %Publication{project: project} = insert(:publication)
+      section = insert(:section, base_project: project)
+      section_resource = insert(:section_resource, %{section: section})
+      Sections.update_section(section, %{
+        root_section_resource_id: section_resource.id
+      })
+
+      {:ok, view, _html} = live(conn, @live_view_lms_instructor_route)
+
+      view
+      |> element("a[phx-value-id=\"product:#{section.id}\"]")
+      |> render_click()
+
+      assert_redirected(view, Routes.delivery_path(OliWeb.Endpoint, :index))
+      assert Sections.get_section_by(%{blueprint_id: section.id})
+    end
+  end
+
+  defp details_view(%Section{type: :blueprint} = section),
+    do: Routes.live_path(OliWeb.Endpoint, OliWeb.Products.DetailsView, section.slug)
+
+  defp details_view(section),
+    do: Routes.project_path(OliWeb.Endpoint, :overview, section.project.slug)
+
+  defp instructor_conn(%{conn: conn}) do
```

Comment:
What about moving this method and the one below to the `Oli.TestHelpers` module?